sdv/single_table/base.py

```diff
@@ -752,6 +752,7 @@ def add_cag(self, patterns):
             try:
                 self.metadata = pattern.get_updated_metadata(self.metadata)
                 self._chained_patterns.append(pattern)
+                self._fitted = False
```

Comment:
Discovered this while upgrading some tests, `preprocess` was erroring after a fitted synthesizer added a new constraint because the constraint wasn't fitted. By setting `_fitted` to False here, we force refitting the constraints during `preprocess`.